src/renderer/api/kube-object.ts

```diff
@@ -40,6 +40,26 @@ export interface IKubeObjectMetadata {
   }[];
 }
 
+export interface IKubeStatus {
+  kind: string;
+  apiVersion: string;
+  code: number;
+  message?: string;
+  reason?: string;
+}
+
+export class KubeStatus implements IKubeStatus {
+  public readonly kind: string;
+  public readonly apiVersion: string;
+  public readonly code: number;
+  public readonly message: string;
+  public readonly reason: string;
+
+  constructor(data: IKubeStatus) {
```

Comment:
This will allow to avoid type check from raw data on constructing object.

```suggestion
  constructor(data: IKubeStatus | any) {
```

Reply:
I disagree that we should do this as it would decrease the benefit of typescript and typing in general. If you don't want to write the type checking specifically the a static function that takes `any` and does the type checking using functions in #2003.